### PR TITLE
Pytorch example

### PR DESCRIPTION
**Pull request by kshitij12345**
# Description

Include a summary of the changes and the related issue.

__Related to:__ `(ClickUp/JIRA task name)`

__Any expected test failures?__

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [x] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Precommit installed and run before pushing changes
- [ ] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [x] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS: Ubuntu
- Python version: 3.9
- Neptune version: 1.1.1
- Affected libraries with version: Pytorch 2.0

### CODE REVIEW

File: integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "i36OvZAgvxkj"
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "! pip install neptune neptune-pytorch torch torchvision numpy torchviz\n"
```
Comment (kshitij12345) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:59]: Will replace `neptune neptune-pytorch` with `neptune[pytorch]` once neptune 1.1.2 (or next version) is released.

File: integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+    "To find your API token and project name, [log in to Neptune](https://app.neptune.ai/).\n",
+    "- In the top-right corner, click your avatar and select **Get your API token**.\n",
+    "- To find and copy your project name, navigate to the project, then click **Settings** → **Properties**.\n",
+    "\n",
+    "### Haven't registered yet?\n",
+    "\n",
+    "To log anonymously to a public project, replace the code below with the following:\n",
+    "\n",
+    "```python\n",
+    "import neptune.new as neptune\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:88]: I've updated the Notebook template, sorry about that.

```suggestion
    "- In the bottom-left corner, expand your user menu and select **Get your API token**.\n",
    "- To copy the project name, in the top-right corner, open the settings menu and select **Properties**.\n",
    "\n",
    "### Haven't registered yet?\n",
    "\n",
    "To log anonymously to a public project, replace the code below with the following:\n",
    "\n",
    "```python\n",
    "import neptune\n",
```

File: integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+   "execution_count": null,
+   "metadata": {
+    "id": "XMdpj-Se4t0U",
+    "tags": [
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "import neptune as neptune\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:105]:
```suggestion
    "import neptune\n",
```

File: integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from neptune.utils import stringify_unsupported\n",
+    "\n",
+    "# (Neptune) `base_namespace` attribute on logger can be used to log metadata consistently\n",
+    "# under the `base_namespace`.\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:301]:
```suggestion
    "# (Neptune) The base_namespace attribute of the logger can be used to log metadata consistently\n",
    "# under the "base_namespace" namespace.\n",
```

Comment (normandy7) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:301]:
```suggestion
    "# (Neptune) The base_namespace attribute of the logger can be used to log metadata consistently\n",
    "# under the 'base_namespace' namespace.\n",
```

File: integrations-and-supported-tools/pytorch/scripts/Neptune_Pytorch.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
+import neptune as neptune
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/scripts/Neptune_Pytorch.py:1]:
```suggestion
import neptune
```

File: integrations-and-supported-tools/pytorch/scripts/Neptune_Pytorch.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+]
+
+# (Neptune) Create NeptuneLogger
+npt_logger = NeptuneLogger(
+    run, model=model, log_model_diagram=True, log_gradients=True, log_parameters=True, log_freq=30
+)
+
+# (Neptune) Log hyperparams
+# NOTE: `base_namespace` attribute on logger can be used to log metadata consistently
+# under the `base_namespace`.
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/scripts/Neptune_Pytorch.py:98]:
```suggestion
# NOTE: The base_namespace attribute of the logger can be used to log metadata consistently
# under the 'base_namespace' namespace.
```

File: integrations-and-supported-tools/pytorch/scripts/requirements.txt [hunk 7]
```diff
@@ -1,4 +1,6 @@
 neptune
+neptune-pytorch
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/scripts/requirements.txt:2]: Nitpick, but should it be:

```suggestion
neptune[pytorch]
```

**Review by normandy7 — CHANGES_REQUESTED** [review]

File: integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb [hunk 8]
```diff
@@ -85,7 +85,7 @@
     "To log anonymously to a public project, replace the code below with the following:\n",
     "\n",
     "```python\n",
-    "import neptune.new as neptune\n",
+    "import neptune\n",
     "\n",
     "run = neptune.init_run(project=\"common/quickstarts\", api_token=neptune.ANONYMOUS_API_TOKEN)\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:90]: Sorry, noticed one more thing.

```suggestion
    "run = neptune.init_run(project=\"common/pytorch-integration\", api_token=neptune.ANONYMOUS_API_TOKEN)\n",
```

**Review by normandy7 — APPROVED** [review]

File: integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb [hunk 9]
```diff
@@ -0,0 +1,472 @@
...
+    "\n",
+    "### Haven't registered yet?\n",
+    "\n",
+    "To log anonymously to a public project, replace the code below with the following:\n",
+    "\n",
+    "```python\n",
+    "import neptune\n",
+    "\n",
+    "run = neptune.init_run(project=\"common/pytorch-integration\", api_token=neptune.ANONYMOUS_API_TOKEN)\n",
+    "```"
```
Comment (normandy7) [inline integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch.ipynb:91]: We can leave this section out, as the code cell has the anonymous/public logging by default.

**Review by normandy7 — CHANGES_REQUESTED** [review]
Sorry, realized the initialization section needed some updating.

**Review by SiddhantSadangi — APPROVED** [review]

**Pull request merged**